packages/beacon-node/src/chain/rewards/blockRewards.ts

```diff
@@ -0,0 +1,193 @@
+
+  return {
+    proposerIndex: block.proposerIndex,
+    total,
+    attestations: blockAttestationReward,
+    syncAggregate: syncAggregateReward,
+    proposerSlashings: blockProposerSlashingReward,
+    attesterSlashings: blockAttesterSlashingReward,
+  };
+}
+
+/**
+ * TODO: Calculate rewards received by block proposer for incuding attestations.
+ */
+function computeBlockAttestationRewardPhase0(
+  _block: phase0.BeaconBlock,
+  _state: CachedBeaconStatePhase0
+): SubRewardValue {
+  throw new Error("Unsupported fork! Block attestation reward calculation is not yet available in phase0");
+}
+
+/**
+ * Calculate rewards received by block proposer for incuding attestations since Altair. Mimics `processAttestationsAltair()`
+ */
+function computeBlockAttestationRewardAltair(
+  block: altair.BeaconBlock,
+  state: CachedBeaconStateAltair
+): SubRewardValue {
+  const {epochCtx} = state;
+  const {effectiveBalanceIncrements} = epochCtx;
+  const stateSlot = state.slot;
+  const rootCache = new RootCache(state);
+  const currentEpoch = epochCtx.epoch;
+  const fork = state.config.getForkSeq(block.slot);
+
+  let blockAttestationReward = 0;
+
+  for (const attestation of block.body.attestations) {
+    const data = attestation.data;
+
+    const committeeIndices = epochCtx.getBeaconCommittee(data.slot, data.index);
+    const attestingIndices = attestation.aggregationBits.intersectValues(committeeIndices);
+
+    const inCurrentEpoch = data.target.epoch === currentEpoch;
+    const epochParticipation = inCurrentEpoch ? state.currentEpochParticipation : state.previousEpochParticipation;
+
+    const flagsAttestation = getAttestationParticipationStatus(
+      fork,
+      data,
+      stateSlot - data.slot,
+      epochCtx.epoch,
+      rootCache
+    );
+
+    let totalBalanceIncrementsWithWeight = 0;
+    for (const index of attestingIndices) {
+      const flags = epochParticipation.get(index);
+      epochParticipation.set(index, flagsAttestation);
+      const flagsNewSet = ~flags & flagsAttestation;
+
+      // Spec:
+      // baseReward = state.validators[index].effectiveBalance / EFFECTIVE_BALANCE_INCREMENT * baseRewardPerIncrement;
+      // proposerRewardNumerator += baseReward * totalWeight
+      let totalWeight = 0;
+      if ((flagsNewSet & TIMELY_SOURCE) === TIMELY_SOURCE) totalWeight += TIMELY_SOURCE_WEIGHT;
+      if ((flagsNewSet & TIMELY_TARGET) === TIMELY_TARGET) totalWeight += TIMELY_TARGET_WEIGHT;
+      if ((flagsNewSet & TIMELY_HEAD) === TIMELY_HEAD) totalWeight += TIMELY_HEAD_WEIGHT;
+
+      if (totalWeight > 0) {
+        totalBalanceIncrementsWithWeight += effectiveBalanceIncrements[index] * totalWeight;
+      }
+    }
+
+    const totalIncrements = totalBalanceIncrementsWithWeight;
+    const proposerRewardNumerator = totalIncrements * state.epochCtx.baseRewardPerIncrement;
+    blockAttestationReward += Math.floor(proposerRewardNumerator / PROPOSER_REWARD_DOMINATOR);
+  }
+
+  return blockAttestationReward;
```

Comment:
there are a lot of code duplicate to `processAttestationsAltair`, I'd go with refactoring that function in a way that we can consume it from here if possible

Reply:
> there are a lot of code duplicate to `processAttestationsAltair`, I'd go with refactoring that function in a way that we can consume it from here if possible

Thanks